README.md

```diff
@@ -11,16 +11,14 @@ This template provisions an HA cluster with ICP 2.1.0.3 enterprise edition.
 ### Pre-requisites
 
 * Working copy of [Terraform](https://www.terraform.io/intro/getting-started/install.html)
-* The example assumes the VMs are provisioned from a template that has ssh public keys loaded in `${HOME}/.ssh/authorized_keys`. After VM creation, terraform will SSH into the VM to prepare and start installation of ICP using the SSH private key provided. If your VM template uses a different user from root, update the `ssh_user` section in [variables.tf](variables.tf#L154)
+* The example assumes the VMs are provisioned from a template that has ssh public keys loaded in `${HOME}/.ssh/authorized_keys`. After VM creation, terraform will SSH into the VM to prepare and start installation of ICP using the SSH private key provided. If your VM template uses a different user from root, update the `ssh_user` section in [variables.tf](variables.tf#L264)
```

Comment:
@jesusmah correct. So, with most cloud providers, when you get a VM they automatically create an ssh key for you, which you can use to ssh into the machine. In the AWS one (and I assume the same applies for IBM Cloud), since you get this ssh key, the Terraform can just use it do the ICP install since it requires password-less ssh access.

You don't get an ssh key with VMWare unless it's already built into the templates. So, instead of having the user manually create ssh keys and set them up in the images (which is very error prone), we have Terraform create and manage the ssh keys for you. However, Terraform needs at least a username/password to ssh into the machines. So it's `pasword-based` ssh to enable the `pasword-less` ssh that's needed by ICP. Weird but that's the only thing I came up with given the VMWare and Terraform limitations.

Reply:
You're correct and hence my comment on not needing templates with ssh keys in them but `password-based` access enabled and then let Terraform do its magic. Does it make sense what I say?

Reply:
Just to align thinking here, the images setup and controlled by the USER do not need any SSH keys setup prior to `terraform apply`.  Terraform will take care of everything required for SSH-based communication during setup.  The USER only has to provide username and password for the base image.

Is that correct? If it is, we don't need to keep anything related to setting up the image with pre-authorized keys (even if that is the way other cloud platforms work), except to say this is a functional change from the previous `2.1.X` stream.

Reply:
@jesusmah @osowski you are both correct. This was something I was working on since `2.1.X` but the previous PR for this never got merged cuz I had a ton other things with it. This is my second attempt at it while taking `3.1` into account.